src/net.rs

```diff
@@ -85,6 +86,27 @@ pub(crate) async fn load_connection_timestamp(
     Ok(timestamp)
 }
 
+/// Returns timestamp of the most recent successful connection
```

Comment:
`load_connection_timestamp()` also needs documentation. But maybe join them and make `addr` an `Option`? Because "by_hostname" isn't literally accurate